parlai/core/metrics.py

```diff
@@ -98,6 +109,14 @@ def __lt__(self, other: Any) -> bool:
         else:
             return self.value() < other
 
+    def __sub__(self, other: Any) -> float:
+        """
+        Used heavily for assertAlmostEqual.
+        """
+        if not isinstance(other, float):
+            raise TypeError('Metrics.__sub__ is intentionally limited to floats.')
```

Comment:
why? is this purely because of the `assertAlmostEqual`?

Reply:
Yup!